src/main/java/duke/Task.java

```diff
@@ -0,0 +1,112 @@
+package duke;
+
+import java.time.LocalDateTime;
+
+/**
+ * Represents a task.
+ */
+public class Task {
+
+    private String status;
+    private String type;
+    private String name;
+    private LocalDateTime time;
+    private Duke duke = new Duke();
+
+    /**
+     * Constructs a task.
+     */
+    public Task() {
+    };
+
+    /**
+     * Constructs a task given a name.
+     *
+     * @param name the task name.
+     */
+    public Task(String name) {
+        this.status = "[ ]";
+        this.name = name;
+    }
+
+    /**
+     * Returns a string representation of the type of the task.
+     *
+     * @return the task type.
+     */
+    public String getType() {
+        return this.type;
+    }
+
+    /**
+     * Returns a string representation of the name of the task.
+     *
+     * @return the task name.
+     */
+    public String getName() {
+        return this.name;
+    }
+
+    /**
+     * Returns a string representation of the status of the task.
+     *
+     * @return the task status.
+     */
+    public String getStatus() {
+        return this.status;
+    }
+
+    /**
+     * Returns the time of the task.
+     *
+     * @return the task time.
+     */
+    public LocalDateTime getTime() {
+        return this.time;
+    }
+
+    /**
+     * Sets the status of the task as specified.
+     *
+     * @param status the status of the task. [X] for done, [ ] for undone.
+     */
+    public void setStatus(String status) {
```

Comment:
Would it be better to use a boolean for the status and then you parse that boolean value into "[ ]" or "[X]"? Just in case someone ends up calling this method with some other random string